### PR TITLE
Add section for hash syntax of enums

### PR DESCRIPTION
**Pull request by tejasbubane**
Closes https://github.com/rubocop-hq/rails-style-guide/issues/144

Ref: https://github.com/rubocop-hq/rubocop-rails/issues/78

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -385,6 +385,24 @@ class Transaction < ActiveRecord::Base
 end
 ----
 
+=== Enums [[enums]]
+
+Prefer hash syntax over arrays for `enums`.
```
Comment (pirj) [inline README.adoc]: WDYT of
```
`enum`s
```
or 
Prefer using the hash syntax for `enum`.

File: README.adoc [hunk 2]
```diff
@@ -385,6 +385,24 @@ class Transaction < ActiveRecord::Base
 end
 ----
 
+=== Enums [[enums]]
```
Comment (pirj) [inline README.adoc]: There's this example:
```ruby
# Rails 4+ enums after attr macros, prefer the hash syntax
```

Do you think it becomes redundant with this new guideline?

File: README.adoc [hunk 3]
```diff
@@ -385,6 +385,24 @@ class Transaction < ActiveRecord::Base
...
+[source,ruby]
+----
+class Transaction < ActiveRecord::Base
+  # bad - implicit values - ordering in array matters
+  enum type: %i[credit debit]
+
+  # good - explicit values - ordering does not matter
+  enum type: {
+    credit: 0,
+    debit: 1,
```
Comment (pirj) [inline README.adoc]: Leading comma.

File: README.adoc [hunk 4]
```diff
@@ -385,6 +385,24 @@ class Transaction < ActiveRecord::Base
...
 ----
 
+=== Enums [[enums]]
+
+Prefer hash syntax over arrays for `enums`.
+
+[source,ruby]
+----
+class Transaction < ActiveRecord::Base
+  # bad - implicit values - ordering in array matters
```
Comment (pirj) [inline README.adoc]: Can't come up with better wording, do you think that "ordering" is sufficient to emphasize that if something is inserted or removed in the middle, things will break, not only when entries are rearranged?

File: README.adoc (hunk 2 above)
Comment (tejasbubane) [reply README.adoc]: Yes, I saw the guideline exists (hence placed this rule just above). I feel the comment reiterates the rule & don't see much of an issue. I can remove `, prefer the hash syntax` if required.

File: README.adoc (hunk 1 above)
Comment (tejasbubane) [reply README.adoc]: Changed to `Prefer using the hash syntax for enum.`

File: README.adoc (hunk 2 above)
Comment (tejasbubane) [reply README.adoc]: Removed.

File: README.adoc (hunk 3 above)
Comment (tejasbubane) [reply README.adoc]: Removed.

File: README.adoc (hunk 4 above)
Comment (tejasbubane) [reply README.adoc]: Added more details to the rule description.

File: README.adoc [hunk 5]
```diff
@@ -385,6 +385,26 @@ class Transaction < ActiveRecord::Base
 end
 ----
 
+=== Enums [[enums]]
```
Comment (koic) [inline README.adoc]: I investigated whether `=== Enums` or `=== Enum` would be better. The API says `Enum`, but the guide says `Enums`, so I think that `Enums` would be fine.

- Guide ... https://guides.rubyonrails.org/active_record_querying.html#enums
- API ... https://api.rubyonrails.org/v5.2.3/classes/ActiveRecord/Enum.html

**Pull request merged**